### PR TITLE
ddl, session: let ddl related tables controlled by `split-table`

### LINKED ISSUE #38346: ddl related tables are not controlled by `split-table` and `enable-cross-table-merge`
When set `split-table = true` and `enable-cross-table-merge = false` the ddl related tables are still merged in one region.

ddl tables:
```sql
mysql> SHOW TABLE mysql.tidb_ddl_history REGIONS;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS | SCHEDULING_CONSTRAINTS | SCHEDULING_STATE |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
|        68 | t_68_     |         |        69 |               1 | 69    |          0 |             0 |          0 |                    0 |                0 |                        |
  |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
1 row in set (0.00 sec)

mysql> SHOW TABLE mysql.tidb_ddl_job REGIONS;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS | SCHEDULING_CONSTRAINTS | SCHEDULING_STATE |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
|        68 | t_68_     |         |        69 |               1 | 69    |          0 |             0 |          0 |                    0 |                0 |                        |
  |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
1 row in set (0.00 sec)
``` 

other tables:
```sql
mysql> SHOW TABLE mysql.user REGIONS;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS | SCHEDULING_CONSTRAINTS | SCHEDULING_STATE |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
|         4 | t_4_      | t_6_    |         5 |               1 | 5     |          0 |             0 |          0 |                    0 |                0 |                        |
  |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
1 row in set (0.00 sec)

mysql> SHOW TABLE mysql.global_priv REGIONS;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS | SCHEDULING_CONSTRAINTS | SCHEDULING_STATE |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
|         6 | t_6_      | t_8_    |         7 |               1 | 7     |          0 |             0 |          0 |                    0 |                0 |                        |
  |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+------------------------+------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #38346

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Let ddl related tables controlled by `split-table`
```

### CODE REVIEW

**Review by xhebox — COMMENTED** [review]
LGTM, but do we really want wo split DDL tables. I think it needs an agreement.

**Comment by Defined2014** [comment]
> LGTM, but do we really want to split DDL tables. I think it needs an agreement.

For flashback cluster, the tikv only supports lock key ranges by region. If ddl table is combined with other tables, it will be locked and couldn't read ddl job. Also we should split ddl_table before flashback cluster start. @xhebox

File: session/session_test.go [hunk 1]
```diff
@@ -51,3 +55,22 @@ func TestInitMetaTable(t *testing.T) {
 		require.True(t, reflect.DeepEqual(metaInMySQL, metaInTest))
 	}
 }
+
+func TestMetaTableRegion(t *testing.T) {
+	enableSplitTableRegionVal := atomic.LoadUint32(&ddl.EnableSplitTableRegion)
```
[the remark this answers is not in the document]
Comment (Defined2014) [reply session/session_test.go:60]: Not quite understand this comment. (

[the remark this answers is not in the document]
Comment (hawkingrei) [reply session/session_test.go:60]: Just code style. it can be ignored.

**Review by xhebox — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@wjhuang2016 PTAL

File: session/session.go [hunk 2]
```diff
@@ -2746,15 +2746,19 @@ func InitDDLJobTables(store kv.Storage) error {
 		if err != nil {
 			return err
 		}
-		p := parser.New()
-		for _, tbl := range DDLJobTables {
-			id, err := t.GetGlobalID()
-			if err != nil {
-				return errors.Trace(err)
+		if s, ok := store.(kv.SplittableStore); ok && atomic.LoadUint32(&ddl.EnableSplitTableRegion) == 1 {
```
Comment (wjhuang2016) [inline session/session.go:2749]: It's better to use a function to handle these tables, so that we don't need to write the same code for MDL table.

Comment (Defined2014) [reply session/session.go]: Done, PTAL @wjhuang2016

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**